cpp/src/io/parquet/experimental/hybrid_scan_helpers.cpp

```diff
@@ -529,4 +529,85 @@ aggregate_reader_metadata::filter_row_groups_with_bloom_filters(
   return bloom_filtered_row_groups.value_or(all_row_group_indices(row_group_indices));
 }
 
+/**
+ * @brief Converts column named expression to column index reference expression
+ */
+named_to_reference_converter::named_to_reference_converter(
+  std::optional<std::reference_wrapper<ast::expression const>> expr,
+  table_metadata const& metadata,
+  std::vector<SchemaElement> const& schema_tree)
+{
+  if (!expr.has_value()) { return; }
+  // create map for column name.
+  std::transform(metadata.schema_info.cbegin(),
+                 metadata.schema_info.cend(),
+                 thrust::counting_iterator<size_t>(0),
+                 std::inserter(_column_name_to_index, _column_name_to_index.end()),
+                 [](auto const& sch, auto index) { return std::make_pair(sch.name, index); });
+
+  auto const& root = schema_tree.front();
+  std::for_each(thrust::counting_iterator<size_t>(0),
+                thrust::counting_iterator(root.children_idx.size()),
+                [&](int32_t col_idx) {
+                  auto const schema_idx = root.children_idx[col_idx];
+                  _top_level_names.insert({col_idx, schema_tree[schema_idx].name});
+                });
+
+  expr.value().get().accept(*this);
+}
+
+std::reference_wrapper<ast::expression const> named_to_reference_converter::visit(
+  ast::column_reference const& expr)
+{
+  // check if column name is in metadata
+  auto const col_name = _top_level_names[expr.get_column_index()];
+  auto col_index_it   = _column_name_to_index.find(col_name);
+  if (col_index_it == _column_name_to_index.end()) {
+    CUDF_FAIL("Column name not found in metadata");
+  }
+  auto col_index = col_index_it->second;
+  _col_ref.emplace_back(col_index);
+  _converted_expr = std::reference_wrapper<ast::expression const>(_col_ref.back());
+  return std::reference_wrapper<ast::expression const>(_col_ref.back());
+}
+
+names_from_expression::names_from_expression(
+  std::optional<std::reference_wrapper<ast::expression const>> expr,
+  std::vector<std::string> const& skip_names,
+  std::vector<SchemaElement> const& schema_tree)
+{
+  if (!expr.has_value()) { return; }
+
+  _skip_names = std::unordered_set<std::string>{skip_names.cbegin(), skip_names.cend()};
+
+  auto const& root = schema_tree.front();
+  std::for_each(thrust::counting_iterator<size_t>(0),
+                thrust::counting_iterator(root.children_idx.size()),
+                [&](int32_t col_idx) {
+                  auto const schema_idx = root.children_idx[col_idx];
+                  _top_level_names.insert({col_idx, schema_tree[schema_idx].name});
+                });
```

Comment:
Only change in the child class: Make sure to extract the names of column_references in the filter expression to be able to select them